caikit_computer_vision/data_model/object_detection.py

```diff
@@ -0,0 +1,42 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Data structures for text object detection in images."""
+
+# First Party
+from py_to_proto.dataclass_to_proto import Annotated, FieldNumber
+import alog
+
+# Local
+from caikit.core import DataObjectBase, dataobject
+from caikit.interfaces.common.data_model import ProducerId
+
+log = alog.use_channel("DATAM")
+
+@dataobject(package="caikit_data_model.caikit_computer_vision")
+class BoundingBox(DataObjectBase):
+    xmin: Annotated[int, FieldNumber(1)]
+    xmax: Annotated[int, FieldNumber(2)]
+    ymin: Annotated[int, FieldNumber(3)]
+    ymax: Annotated[int, FieldNumber(4)]
+
+@dataobject(package="caikit_data_model.caikit_computer_vision")
+class DetectedObject(DataObjectBase):
+    score: Annotated[float, FieldNumber(1)]
+    label: Annotated[str, FieldNumber(2)]
+    box: Annotated[BoundingBox, FieldNumber(3)]
```

Comment:
are the bounding boxes relative to some position ?

Reply:
Yup, this may be model specific, but in general I'd assume these are image coordinates

![image coordinates](https://documentation.euresys.com/Products/OPEN_EVISION/OPEN_EVISION/en-us/Content/Resources/Images/03-1-2_Manipulating/Image_Coordinate_Systems_integer.png)

Reply:
Should the coordinate system be a part of the DM to make it evident to the user? Or are you planning to add that bit later depending on how the modules evolve ?

Reply:
I think that it should be omitted. In general, the data model (will be) viewable in different common formats, e.g., PIL format, so the way that you index into an image will essentially depend on what you're viewing it as. But I think in general, image coordinates are a pretty standard choice, and this is what PIL images use, which is important since it's the hub backend format.

So the main time it matters is just at serialization time, but even then it should be a hidden detail I think, since it'll be serialized as raw bytes / the image will be flattened anyway

Reply:
Is there a possibility of misinterpretation if the encoding time it was aimed to be a different coordinate system then what users might load it with?

Reply:
If they're taking this to other image formats in a client, maybe, but I don't think it's super likely, and even then, what really matters is that we're good about documenting that everything is currently done in image coordinates, which is pretty standard (e.g., for open cv / numpy array, PIL images, torch tensors etc).

This may become more important later on as we add more background, but for the time being, since the Image data model uses PIL images as a hub format, I'd like to omit the coordinate system. I see your point though - in the future, maybe we could consider adding an inference arg that has some common transformations to convert to alternate coordinate systems, where everything is done in image coordinates by default? 🤔